disco_aws_automation/disco_vpc_endpoints.py

```diff
@@ -0,0 +1,124 @@
+"""
+Create / Destroy VPC endpoints
+"""
+
+import json
+
+import boto3
+
+S3_POLICY = {
+    "Statement": [
+        {
+            "Action": "*",
+            "Effect": "Allow",
+            "Resource": "*",
+            "Principal": "*"
+        }
+    ]
+}
+
+
+class DiscoVPCEndpoints(object):
+    """
+    Create / Destroy VPC endpoints
+    """
+
+    def __init__(self, vpc_id, boto3_ec2_client=None):
+        self.vpc_id = vpc_id
+        self._boto3_ec2 = boto3_ec2_client
+
+    @property
+    def boto3_ec2(self):
+        """
+        Lazily creates boto3 EC2 connection
+        """
+        if not self._boto3_ec2:
+            self._boto3_ec2 = boto3.client('ec2')
+        return self._boto3_ec2
+
+    def service_name(self, service):
+        """
+        Find full, region specific service endpoint name from colloquial
+        name (eg: s3 -> com.amazonaws.us-west-2.s3)
+        """
+        endpoint_responce = self.boto3_ec2.describe_vpc_endpoint_services()
+        service_names = endpoint_responce["ServiceNames"]
+
+        matching_service_names = [
+            service_name
+            for service_name in service_names
+            if service_name.endswith(service)
+        ]
+        return matching_service_names[0]
+
+    def filters(self, service):
+        """
+        Return all endpoints of a particular service in the vpc.
+        """
+        return [
+            {"Name": "vpc-id", "Values": [self.vpc_id]},
+            {"Name": "service-name", "Values": [self.service_name(service)]}
+        ]
+
+    def _all_vpc_route_tables_ids(self):
+        route_table_response = self.boto3_ec2.describe_route_tables(
+            Filters=[{"Name": "vpc-id", "Values": [self.vpc_id]}]
+        )
+        return [
+            rt["RouteTableId"]
+            for rt in route_table_response["RouteTables"]
+        ]
+
+    def update_s3(self, route_table_ids=None, policy_dict=None, dry_run=False):
+        """
+        Creates/updates s3 endpoint and update corresponding route tables
+        """
+        route_table_ids = route_table_ids or self._all_vpc_route_tables_ids()
+        policy_dict = policy_dict or S3_POLICY
+
+        self.delete_s3(dry_run)
+        self.boto3_ec2.create_vpc_endpoint(
+            DryRun=dry_run,
```

Comment:
Calling boto3 API functions with `DryRun` set to True would cause an exception being thrown. Is that what we want here?